bbot/modules/internal/excavate.py

```diff
@@ -1,476 +1,858 @@
-                        reason = f"it exceeds the max number of links per page ({self.web_spider_links_per_page})"
-                    elif exceeds_redirect_distance:
-                        reason = (
-                            f"its spider distance exceeds the max number of redirects ({self.excavate.max_redirects})"
-                        )
-                    self.excavate.debug(f"Tagging {url_event} as spider-danger because {reason}")
-                    url_event.add_tag("spider-danger")
-                if "url" in event.data:
-                    message = f"Found URL [{result}] from parsing [{event.data.get('url')}] with regex [{name}]"
-                else:
-                    message = (
-                        f"Found URL [{result}] from parsing [{event.parent.data.get('path')}] with regex [{name}]"
-                    )
-                self.excavate.debug(message)
-                await self.excavate.emit_event(url_event)
-                if url_in_scope:
-                    urls_found += 1
-
-    async def _search(self, content, event, **kwargs):
-        parsed = getattr(event, "parsed_url", None)
-        for name, regex in self.compiled_regexes.items():
-            # yield to event loop
-            await self.excavate.helpers.sleep(0)
-            for result in await self.helpers.re.findall(regex, content):
-                if name.startswith("full"):
-                    protocol, other = result
-                    result = f"{protocol}://{other}"
-
-                elif parsed and name.endswith("-tag"):
-                    path = html.unescape(result[1])
-
-                    for p in self.prefix_blacklist:
-                        if path.lower().startswith(p.lower()):
-                            self.excavate.debug(
-                                f"omitted result from a-tag parser because of blacklisted prefix [{p}]"
-                            )
-                            continue
+class excavate(BaseInternalModule):
+    """
+    Example (simple) Excavate Rules:
 
-                    if not self.compiled_regexes["fullurl"].match(path):
-                        source_url = event.parsed_url.geturl()
-                        result = urljoin(source_url, path)
-                        # this is necessary to weed out mailto: and such
-                        if not self.compiled_regexes["fullurl"].match(result):
-                            continue
-                    else:
-                        result = path
+    class excavateTestRule(ExcavateRule):
+        yara_rules = {
+            "SearchForText": 'rule SearchForText { meta: description = "Contains the text AAAABBBBCCCC" strings: $text = "AAAABBBBCCCC" condition: $text }',
+            "SearchForText2": 'rule SearchForText2 { meta: description = "Contains the text DDDDEEEEFFFF" strings: $text2 = "DDDDEEEEFFFF" condition: $text2 }',
+        }
+    """
 
-                yield result, name
+    watched_events = ["HTTP_RESPONSE", "RAW_TEXT"]
```

Comment:
Watches for `RAW_TEXT`, but it seems like `handle_event` is only set up to handle `HTTP_RESPONSE`. We should write tests for `RAW_TEXT` (or just remove it for now).

Reply:
I think removing for this PR is the best option. We can add it back in as a complete package, with new tests, etc.
https://github.com/blacklanternsecurity/bbot/pull/1471/commits/661331ff641d65e96cdf1d5379b08bb7402c1746